addons/l10n_mn/migrations/1.1/post-migrate.py

```diff
@@ -0,0 +1,259 @@
+
+    # Now, do the migration!
+    # STEP 1: Remove obsolete tag vat_report_tag11
+    cr.execute('''
+        SELECT res_id
+        FROM ir_model_data
+        WHERE name = 'vat_report_tag11'
+            AND module = 'l10n_mn'
+            AND model = 'account.account.tag'
+    ''')
+    vat_report_tag11_id = cr.fetchall()
+    if vat_report_tag11_id:
+        vat_report_tag11_id = vat_report_tag11_id[0][0]
+        cr.execute('''
+            DELETE FROM account_tax_repartition_financial_tags
+            WHERE account_account_tag_id = %s
+        ''', [vat_report_tag11_id])
+        cr.execute('''
+            DELETE FROM account_account_tag_account_tax_repartition_line_rel
+            WHERE account_account_tag_id = %s
+        ''', [vat_report_tag11_id])
+        cr.execute('''
+            DELETE FROM account_account_tag_account_move_line_rel
+            WHERE account_account_tag_id = %s
+        ''', [vat_report_tag11_id])
+
+    # STEP 2: Unset tax tags on base lines / tax lines that shouldn't be there.
+    cr.execute('''
+        SELECT res_id
+        FROM ir_model_data
+        WHERE
+            module = 'l10n_mn'
+            AND model = 'account.account.tag'
+            AND name IN %s
+    ''', [tuple(tax_tag_changes)])
+    old_tag_ids = [row[0] for row in cr.fetchall()]
+
+    cr.execute('''
+        WITH tag_aml_rel_to_delete AS (
+            SELECT DISTINCT
+                tag_aml_rel.account_account_tag_id,
+                tag_aml_rel.account_move_line_id
+            FROM account_account_tag_account_move_line_rel tag_aml_rel
+            JOIN account_account_tag tag
+                ON tag.id = tag_aml_rel.account_account_tag_id
+            JOIN ir_model_data
+                ON ir_model_data.res_id = tag.id
+                AND ir_model_data.module = 'l10n_mn'
+                AND ir_model_data.model = 'account.account.tag'
+            JOIN account_move_line aml
+                ON aml.id = tag_aml_rel.account_move_line_id
+            LEFT JOIN account_move_line_account_tax_rel amlatr
+                ON amlatr.account_move_line_id = aml.id
+            WHERE
+                tag.id IN %s
+                AND (
+                    ir_model_data.name IN %s
+                        AND aml.tax_line_id IS NOT NULL
+                    OR
+                    ir_model_data.name IN %s
+                        AND amlatr.account_move_line_id IS NOT NULL
+                )
+        )
+        DELETE FROM account_account_tag_account_move_line_rel tag_aml_rel
+        USING tag_aml_rel_to_delete
+        WHERE tag_aml_rel.account_account_tag_id = tag_aml_rel_to_delete.account_account_tag_id
+            AND tag_aml_rel.account_move_line_id = tag_aml_rel_to_delete.account_move_line_id
+    ''', [tuple(old_tag_ids), tuple(tax_tags_for_base_lines), tuple(tax_tags_for_tax_lines)])
+
+    # STEP 3: Change the VAT tags on AMLs, choosing the +tag or -tag where appropriate.
+    cr.execute('''
```

Comment:
Do not we have existing code that can handle such things?  (like in the branch of clbr?)

Reply:
Possibly! I can have a look at that branch. But I'm pretty sure this code works too.